[CDS-76320]: Update run-a-terraform-plan-with-the-terraform-plan-step.md (#3892)
harness/developer-hub: docs/continuous-delivery/cd-infrastructure/terraform-infra/rollback-provisioned-infra-with-the-terraform-rollback-step.md

## Review: Terraform Rollback

> When rollback happens, Harness rolls back the provisioned infrastructure to the previous successful version of the Terraform state and using config files/terraform configuration from latest successful deployment with matching  **Provisioner Identifier**.

Reviewer: ```suggestion
When rollback is performed, Harness rolls back the provisioned infrastructure to the previous successful version of the Terraform state using config files or the Terraform configuration from the latest successful deployment with a matching  **Provisioner Identifier**.
```

However, let's look at the situation where module3 succeeds and now you have module1, module2, and module3 deployed. If the next deployment fails, the rollback will only roll back to the Terraform state with module3 deployed. Module1 and module2 weren't in the previous Terraform state, so the rollback excludes them.

> Rollback is not possible if you run apply with **Skip state storage** option and no terraform backed configured in your terraform files.
> Having Rollback step in such scenario would be a wrong setup and may end-up with an unexpected result.

Reviewer: ```suggestion
Rollback is not possible if you run the Terraform Apply step with the **Skip state storage** option enabled and no Terraform backed is configured in your Terraform files.
Using the Rollback step in such a scenario would be an incorrect setup and might cause an unexpected result.
```
